base/test.jl

```diff
@@ -22,6 +22,30 @@ export GenericString
 
 #-----------------------------------------------------------------------
 
+# Backtrace utility functions
+function ip_matches_func_and_name(ip, func::Symbol, dir::String, file::String)
+    for fr in StackTraces.lookup(ip)
+        if fr === StackTraces.UNKNOWN || fr.from_c
+            return false
+        end
+        path = string(fr.file)
+        fr.func == func && dirname(path) == dir && basename(path) == file && return true
+    end
+    return false
+end
+
+function scrub_backtrace(bt)
+    do_test_ind = findfirst(addr->ip_matches_func_and_name(addr, :do_test, ".", "test.jl"), bt)
```

Comment:
Since this finds the first function in the stacktraces with the given name,  we will find the correct one since it is always above any user function that would be fine. User functions also have the full path in the `.file` field of stack frames so the path would also not match. If you still feel it is necessary I could of course add a some random stuff to the end of the function name.